### DISCUSSION TITLE
FastAPI & RabbitRouter

### DISCUSSION THREAD

**Discussion by JoshJarabek7** [Lancetnik/Propan#133] [Questions] [answered]
First off, I want to say this project is amazing! It's so simplified and way less verbose than aio-pika. So hats off to y'all and thank you for making it.

I do have a question about FastAPI and RabbitRouter.
Can I pass the RabbitMQ connection string to every router or is that going to create multiple brokers? I've been creating a broker on the main.py and a get_broker() function, but I've been running into circular import issues quite due to this. Or is this accessible to the other RabbitRouters via the lifespan state?

Thanks again!

**Comment by Lancetnik** [comment, marked as answer]
@JoshJarabek7, hi! Thanks for your kind feedback!
Unfortunately, I have not provided for the inclusion of different PropanRouter into each other
Will test this behavior and connect you a little bit later.
Thank you for making the project better too!

**Reply by JoshJarabek7** [reply]
Thanks for the quick reply! And sounds good 👍

**Reply by Lancetnik** [reply]
I implemented Routers nesting in the V2
Please, wait a bit (about 2 weeks) and we will release it

For now, you can nest Routers via lifespans
```python
router1 = RabbitRouter()
router2 = RabbitRouter(lifespan=router1.lifespan_context)
app = FastAPI(lifespan=router2.lifespan_context)
```
But it will create a new connection for each router

**Reply by JoshJarabek7** [reply]
Thank you!
